wordpress-mobile/WordPress-Android
# [Tooling] Remove duplicate locales in translations download

The list of `ALL_LOCALES` — that we use to derive the list of locales to download `strings.xml` files and Play Store metadata `*.txt` files from GlotPress — contained duplicates, and the list was also in some random/arbitrary order.

ℹ️  Note: this PR targets `release/20.0` because the next time we'll use this will be during next beta or final release.

I figured that, given that this change reduces the number of network requests to GlotPress (by removing useless duplicate calls), and the fact that we've been having the occasional `429 Too Many Requests` by GlotPress lately, better profit from the risk reduction brought by this PR sooner rather than later — especially if it reduces the risk for @ParaskP7 to get missing translations when he'll cover for me on submission day next Friday.

## What

In this PR:

 - fa13640: Removes locales that were duplicated in the list.
    - The duplicates led to those locales being downloaded twice from GlotPress when exporting translations, which was not great especially given we lately get some rate limiting once in a while from GlotPress already.
 - 106ff60: Reorders the list of locales so that it's kept alphabetical.
    - The list still keeps the locales used for "both in-app `strings.xml` and release notes / Play Store metadata" first, and the list of additional locales used "only for in-app `strings.xml` but not release notes" next, to make it easier to grasp which locales are used only for metadata (filtered into `RELEASE_NOTES_LOCALES` [a couple of lines below](https://github.com/wordpress-mobile/WordPress-Android/blob/919b78405c2f430d7b0a5a5b14e15a9035d554ea/fastlane/lanes/localization.rb#L65-L67)).
    - But within each of those 2 subsets, locales have been ordered alphabetically for consistency.
 - 0d4b74c: Updated the comment to explain a bit more about that definition of this `ALL_LOCALES` array constant.

## To Test

 - Creating a test branch from this one
 - Run the `download_metadata_strings` lane, and check that it still downloads all the same list of locales for release notes as before
 - Run `download_translations`, and check that it still downloads all the same list of locales for in-app strings as before

That being said, I think that just carefully inspecting the diff would be enough testing, since it's only removal of duplicates and reordering of the list.

### Diff tip: use `--color-moved`

You can checkout this branch then run `git diff --color-moved=plain release/20.0` to get a colored diff that shows more clearly which lines were just moved from one location (pink) to another (cyan), as opposed to the ones completely deleted (red) or added (green):
<img width="1397" alt="image" src="https://user-images.githubusercontent.com/216089/171839920-3fa76b7b-f624-4c0a-880d-e4a4613b0595.png">

You can then check that the only locales that were deleted (red) were ones already present higher in the file and list.
